### DISCUSSION TITLE
Talking about scaling

### DISCUSSION THREAD

**Discussion by somehowchris** [sorry-cypress/sorry-cypress#501] [Q&A]
So...... a topic which never gets old.

First of all, awesome project, love it. Now I would like to use it for further things. I searched for parameters to scale or things that don't scale.

I couldn't find a single word about ressource usage (CPU, RAM, ....) and scaling the apps. So whats up?

__Edit__: @agoldis ?

**Comment by agoldis** [comment]
Is there anything specific in terms of scaling you're looking for?

The runtime configuration is not demanding, the bottleneck is the storage in most cases.

**Reply by somehowchris** [reply]
Yes:
- ever ran a test with 2k enigeers and managers accessing stuff?
- can the director service be scaled?
- do they scale horizontally or vertically?
- do you plan on integrating devops helping tools (currently on my mind, apm, sentry, prometheus, ....)
- Will you publish arm based docker images? (docker buildx as a reference)
- ever though about using a more efficient language?
- Do you run any cron jobs? (cron job triggers aren't supposed to be placed within apps, doesn't scale)
- How about authentication and authorization?


Got many more, but that's a beginning

**Reply by somehowchris** [reply]
I know sounds real stupid, do you have a architecture diagram, containing the api, director, dashboard, dependent services, agents and users accessing it? Couldn't find one. Cause if there needs to be access to the dependent services, sadly it would mess with network configurations and wouldn't be ready for my case

**Comment by agoldis** [comment]
- ever ran a test with 2k engineers and managers accessing stuff?
It's quite hard to find 2k engineers and managers to test an open-source project, so - no :) I know that there are big organizations running sorry-cypress, not sure how big is the actual scale - i.e. an org can be big, but specific department using SC may be just a few engineers.

- can the director service be scaled?
Yes but MongoDB will be a bottleneck. I have faced some performance / scale problems while working on currents.dev and had to change the infrastructure. however, for open-source version it would increase the complexity and price of installation, so I decided to keep it simple.

- do they scale horizontally or vertically?
horizontally

- do you plan on integrating devops helping tools (currently on my mind, apm, sentry, prometheus, ....)
no,  but believe using k8s would allow integrating those 

- Will you publish arm based docker images? (docker buildx as a reference)
if needed, one can build any docker image from the source code - it's all in open access

- ever though about using a more efficient language?
yes, I thought. that's a deeper conversation though... 

- Do you run any cron jobs? (cron job triggers aren't supposed to be placed within apps, doesn't scale)
not a cron jobs per-se, but there're some in-app periodic tasks for checking runs timeouts

- How about authentication and authorization?
Not supported in sorry-cypress


> I know sounds real stupid, do you have a architecture diagram, containing the api, director, dashboard, dependent services, agents and users accessing it? Couldn't find one. Cause if there needs to be access to the dependent services, sadly it would mess with network configurations and wouldn't be ready for my case

There's no such diagram, you can infer the architecture from cloudfront or helm charts.

If you have specific questions / use case - let's discuss it. https://calendly.com/agoldis/30
